frame/balances/README.md

```diff
@@ -0,0 +1,124 @@
+Modifications to the original Balances pallet are added.
```

Comment:
Can you list the changes ?
- So that builders know what is added
- So that it will be easier to uplift for us